smst.go

```diff
@@ -9,8 +9,14 @@ import (
 )
 
 const (
-	// The number of bits used to represent the sum of a node
+	// The number of bytes used to represent the sum of a node
 	sumSizeBytes = 8
+
+	// The number of bytes used to track the count of non-empty nodes in the trie.
+	// TODO_TECHDEBT: Since we are using sha256, we could theoretically have
+	// 2^256 leaves. This would require 32 bytes, and would not fit in a uint64.
+	// For now, we are assuming that we will not have more than 2^64 leaves.
+	countSizeBytes = 8
```

Comment:
Ref: As we are using big endian this can be calculated on the fly using [1]. If we don't hard code this value but instead do
```
countBz := data[spec.ph.PathSize()+sumSizeBytes:]
count := new(big.Int).SetBytes(countBz) // this is a *big.Int holding the remaining value
```

[1] https://pkg.go.dev/math/big#Int.SetBytes